src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -95,15 +103,27 @@ static HttpWebRequestActivitySource()
         }
     }
 
-    internal static HttpClientInstrumentationOptions Options
+    internal static HttpClientInstrumentationOptions TracingOptions
     {
-        get => options;
+        get => tracingOptions;
         set
         {
-            options = value;
+            tracingOptions = value;
 
-            emitOldAttributes = value.HttpSemanticConvention.HasFlag(HttpSemanticConvention.Old);
-            emitNewAttributes = value.HttpSemanticConvention.HasFlag(HttpSemanticConvention.New);
+            tracingEmitOldAttributes = value.HttpSemanticConvention.HasFlag(HttpSemanticConvention.Old);
+            tracingEmitNewAttributes = value.HttpSemanticConvention.HasFlag(HttpSemanticConvention.New);
+        }
+    }
+
+    internal static HttpClientMetricInstrumentationOptions MetricsOptions
+    {
+        get => metricsOptions;
+        set
+        {
+            metricsOptions = value;
+
+            metricsEmitOldAttributes = value.HttpSemanticConvention.HasFlag(HttpSemanticConvention.Old);
+            metricsEmitNewAttributes = value.HttpSemanticConvention.HasFlag(HttpSemanticConvention.New);
         }
     }
```

Comment:
The fact that old & new attributes is separate for Tracing and Metrics seems wrong to me. The HttpSemanticConvention comes from an environment variable. It should be set once for the whole process.

Reply:
Ya it is weird. I think though if our plan is to remove the new/old/dupe stuff soon we should just live with it for a bit 😄

Having the two does allow you to do some interesting things. Imagine this...

```csharp
        using var tracerProvider = Sdk.CreateTracerProviderBuilder()
            .AddHttpClientInstrumentation()
            .AddOtlpExporter()
            .ConfigureServices(s => s.AddSingleton<IConfiguration>(new ConfigurationBuilder()
                .AddInMemoryCollection(new Dictionary<string, string> { ["OTEL_SEMCONV_STABILITY_OPT_IN"] = "http" })
                .Build()))
            .Build();

        using var meterProvider = Sdk.CreateTracerProviderBuilder()
            .AddHttpClientInstrumentation()
            .AddPrometheusExporter()
            .ConfigureServices(s => s.AddSingleton<IConfiguration>(new ConfigurationBuilder()
                .AddInMemoryCollection(new Dictionary<string, string> { ["OTEL_SEMCONV_STABILITY_OPT_IN"] = "http/dup" })
                .Build()))
            .Build();
```

What that would do is send "new" semantic conventions via OTLP for traces and send "dupe" semantic conventions via Prometheus for metrics. That is because we support IConfiguration for each provider not just envvars for a process.

Reply:
> I think though if our plan is to remove the new/old/dupe stuff soon we should just live with it for a bit 😄

👍